### PR TITLE
feat(): add adapter option mongoDriverOpts, add default option ignoreUndefined

### LINKED ISSUE #30: Undefined properties are set to null in Mongodb
## Description

versions :
js-data: **3.0.0-rc.6**
js-data-mongodb : **1.0.0-rc.1**
## Steps to reproduce

Here is my model schema : 

```
{
  "type": "object",
  "properties": {
    "a": {
      "type": "string"
    },
    "b": {
      "type": "string"
    }
  },
  "required": ["a"],
  "additionalProperties": false
}
```

When i insert or update not all the properties of my model (example with only property "a" : `{a: "a"}`), the others properties are set to `undefined` by js-data. So the input `props` of methods `create` or `update` looks like : 

```
{
  a: "a",
  b: undefined
}
```

The problem is that the Node.js mongodb driver, insert `null` in the database for all the `undefined` properties. So when you just want to update property "a", the previous value of property "b" is erase by a `null` value.

The workaround solution i found, is to remove all undefined properties of the object in methods `beforeCreate` and `beforeUpdate` on the adapter. But i think it could be made directly on insert or update methods.

``` javascript
// Workaround to avoid undefined values to be set to null in MongoDB
const adapter = new MongoDBAdapter({
  uri: config.MONGODB_URI,
  beforeUpdate(mapper, id, props) {
    const removeUndefined = (obj) => {
      Object.keys(obj).forEach(key =>
        (obj[key] && typeof obj[key] === 'object') && removeUndefined(obj[key]) ||
        (obj[key] === undefined) && delete obj[key]
      );
      return obj;
    };
    removeUndefined(props);
  }
});
```

Solution can be found here : http://stackoverflow.com/questions/286141/remove-blank-attributes-from-an-object-in-javascript. It should be recursive and only remove undefined properties.

Any thoughts ?

Thanks!

### PR DESCRIPTION
**Pull request by crobinson42**
Fixes #30 
- [x] - If applicable, appropriate JSDoc comments were updated in source code
- [ ] - If applicable, appropriate changes to js-data.io docs have been suggested ("Suggest Edits" button)
- [x] - Unit test coverage did not decrease & appropriate tests changes made

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -32,7 +32,18 @@ const DEFAULTS = {
...
+
+  /**
+   * MongoDB Driver options
+   *
+   * @name MongoDBAdapter#mongoDriverOpts
+   * @type {object}
+   * @default { ignoreUndefined: true }
+   */
+  mongoDriverOpts: {
+    ignoreUndefined: true
```
Comment (danivek) [inline src/index.js:45]: I think you have missed the db property:

```
{
  "db": {
    "ignoreUndefined": true
  }
}
```

Comment (crobinson42) [reply src/index.js:45]: @danivek Depends on which doc you're looking at.. they both are accepted.

ref.:
[mongoclient.connect](http://mongodb.github.io/node-mongodb-native/2.1/reference/connecting/connection-settings/)
[mongoclient](https://mongodb.github.io/node-mongodb-native/api-generated/mongoclient.html)

**Comment by jmdobry** [comment]
Test?

**Comment by crobinson42** [comment]
@jmdobry g2g

**Review by jmdobry — APPROVED** [review]

**Review by jmdobry — APPROVED** [review]

**Pull request merged**